SitumWayfinding/Classes/Positioning/PositioningViewController.swift

```diff
@@ -1308,4 +1315,54 @@ extension PositioningViewController {
     }
 }
 
+extension PositioningViewController {
+    func prepareMarkerStairs() {
+        let icon = self.imageWithImage(
+            image: UIImage(named: "change_floor")!,
+            scaledToSize: CGSize(width: 40.0, height: 40.0)
+        )
+        let markerView = UIImageView(image: icon)
+        self.stairs.iconView = markerView
+        self.stairs.position = CLLocationCoordinate2D(latitude: 0, longitude: 0)
+        self.stairs.map = nil
 
+    }
+    
+    func showMarkerStairs(position: CLLocationCoordinate2D) {
+        self.stairs.position = position
+        self.stairs.map = self.mapView
+    }
+    
+    func imageWithImage(image:UIImage, scaledToSize newSize:CGSize) -> UIImage{
+        UIGraphicsBeginImageContextWithOptions(newSize, false, 0.0)
+        image.draw(in: CGRect(x: 0, y: 0, width: newSize.width, height: newSize.height))
+        let newImage:UIImage = UIGraphicsGetImageFromCurrentImageContext()!
+        UIGraphicsEndImageContext()
+        return newImage
+    }
+    
+    func checkChangeFloor(segments: Array<SITRouteSegment>, selectedFloor: SITFloor) {
+        self.stairs.map = nil
+        for segment in segments {
+            if segment.floorIdentifier == selectedFloor.identifier &&
+                self.lastSelectedMarker?.floorIdentifier != selectedFloor.identifier {
+                if segments.count > 1 {
```

Comment:
Este if non debería ir xa fora do for, se non hai segmentos, para que facer todo o anterior?